nautobot/extras/api/serializers.py

```diff
@@ -653,6 +655,27 @@ class Meta:
         ]
         opt_in_fields = ["computed_fields"]
 
+    def validate(self, data):
+        has_sensitive_variables = data.get("has_sensitive_variables")
+        approval_required = data.get("approval_required")
+
+        # note no validation for on creation of jobs because we do not support user creation of Job records via API
+        if self.instance:
+            if (
+                (self.instance.has_sensitive_variables and "has_sensitive_variables" not in data)
+                or has_sensitive_variables is True
+            ) and approval_required is True:
+                raise serializers.ValidationError(
+                    {"approval_required": ["A job with sensitive variables cannot be marked as requiring approval"]}
+                )
+            elif (
+                (self.instance.approval_required and "approval_required" not in data) or approval_required is True
+            ) and has_sensitive_variables is True:
+                raise serializers.ValidationError(
+                    {"has_sensitive_variables": ["A job marked for approval cannot have sensitive variables"]}
+                )
```

Comment:
Would something like this work basically as well and be simpler? Or do we have to handle the data variables separately from the instance variables?

```suggestion
        # note no validation for on creation of jobs because we do not support user creation of Job records via API
        if self.instance:
            has_sensitive_variables = data.get("has_sensitive_variables", self.instance.has_sensitive_variables)
            approval_required = data.get("approval_required", self.instance.approval_required)
            
            if approval_required and has_sensitive_variables:
                raise serializers.ValidationError(
                    {
                        "approval_required": [
                            "A job with sensitive variables cannot also be marked as requiring approval"
                        ],
                        "has_sensitive_variables": [
                            "A job with sensitive variables cannot also be marked as requiring approval"
                        ]
                    }
                )
```

Reply:
Updating only one field, and you get error for two fields don't look right IMO